mass_mailing_import_mailing_list/wizard/import_mailing_list.py

```diff
@@ -227,19 +228,41 @@ def import_data(self):
             'target': 'new',
         }
 
-    def parse_csv_data(self):
+    def parse_csv_data(self, extention):
         # Decode data to string, Odoo supplies it as base64
+        delimiter = ';' if extention == '.csv' else '\t'
         csv_data = base64.b64decode(self.file)
-        try:
-            data_file = io.StringIO(csv_data.decode("ISO-8859-1"))
-        except UnicodeDecodeError as e:
-            raise UserError(f'Error {e}')
-        # Read CSV
-        headers, *data = csv.reader(data_file, delimiter=';')
-        if len(data) == 0:
-            raise UserError(f'File is empty')
-        # Verify Header, Force it lowercase and make a dict
-        headers = self.check_header(headers, self.import_type)
+        caught = False
+        for encoding in ('ISO-8859-1', 'cp1252', 'utf-8'):
+            try:
+                # Read CSV
+                data_file = io.StringIO(csv_data.decode(encoding))
+                headers, *data = csv.reader(data_file, delimiter=delimiter)
+                if len(data) == 0:
+                    raise UserError(f'File is empty')
+                # Verify Header, Force it lowercase and make a dict
+                headers = self.check_header(headers, self.import_type)
+            except UnicodeDecodeError:
+                pass
+            except (UserError) as e:
```

Comment:
Is parentheses needed?

Reply:
Not any more, they are if you catch several errors, ill remove